sdv/metadata/multi_table.py

```diff
@@ -333,6 +368,87 @@ def _log_detected_table(single_table_metadata):
         table_json = json.dumps(table_dict, indent=4)
         LOGGER.info(f'Detected metadata:\n{table_json}')
 
+    def _validate_all_tables_connected(self, parent_map, child_map):
+        """Get the connection status of all tables.
+
+        Args:
+            parent_map (dict):
+                Dictionary mapping each parent table to its child tables.
+            child_map (dict):
+                Dictionary mapping each child table to its parent tables.
+
+        Returns:
+            dict specifying whether each table is connected the other tables.
+        """
+        nodes = list(self.tables.keys())
+        if len(nodes) == 1:
+            return
+
+        queue = [nodes[0]]
+        explored = set()
+        connected = {table_name: False for table_name in nodes}
+
+        while queue:
+            node = queue.pop()
+            explored.add(node)
```

Comment:
I assume this was added because it's possible to start on a node with no neighbors right? If so can we add a unit test case for this situation?